third-party/realdds/src/dds-topic-reader.cpp

```diff
@@ -88,10 +88,8 @@ void dds_topic_reader::run( qos const & rqos )
 {
     // The DataReader is the Entity used by the application to subscribe to updated values of the data
     eprosima::fastdds::dds::StatusMask status_mask;
-    if( _on_subscription_matched )
-        status_mask << eprosima::fastdds::dds::StatusMask::subscription_matched();
-    if( _on_data_available )
-        status_mask << eprosima::fastdds::dds::StatusMask::data_available();
+    status_mask << eprosima::fastdds::dds::StatusMask::subscription_matched();
```

Comment:
I liked it better as before, do you think it's problematic?
I see you just call `handle_frames`, so why not set it when you `open()` once and just have it do nothing if not streaming?

Reply:
The listener flags must be set here, not later, or the callbacks (`on_data_available`)  won't be called.

Reply:
That was the whole idea -- you set the callbacks BEFORE opening the reader, not during.

Reply:
But the callback is available to us in `start_streaming` and I want to `run` the reader at open to best mimic USB behavior.
Let's discuss tomorrow with @Nir-Az and decide what behavior we want on open and on start

Reply:
Restored previous state. Setting the callback before reader `run`